charmhub/refresh.go

```diff
@@ -84,8 +84,31 @@ func (c *RefreshClient) Refresh(ctx context.Context, config RefreshConfig) ([]tr
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+
+	// Add Juju-Metadata headers for the charms' unique series and
+	// architecture values, for example:
+	//
+	// Juju-Metadata: series=bionic
+	// Juju-Metadata: arch=amd64
+	// Juju-Metadata: series=focal
+	headers := make(http.Header)
+	seriesAdded := make(map[string]bool)
+	archsAdded := make(map[string]bool)
```

Comment:
We have a juju/collections/set that could be used here

Reply:
Used.